src/plugin/plugin_info.rs

```diff
@@ -9,6 +10,11 @@ pub struct PluginInfo {
 
     pub create_account: Option<extern fn() -> Account>,
     pub destroy_account: Option<extern fn(acc: Account)>,
+    /// Instructs the plugin to post a message in the associated channel.
+    /// The lifetime of msg is only guaranteed during the function call.
+    /// To keep the message for longer (likely required), make a copy.
+    /// TODO: Success or failure indication.
+    pub post_message: Option<extern fn(msg: * const Message)>,
```

Comment:
We can make a return type for this such as enum `Status`

Reply:
What we'll likely end up needing is another function in the API to set the status, because that function is not supposed to be blocking, and in most cases you don't know if it had succeeded or failed until well after this function had completed.